npm/pod_test.go

```diff
@@ -202,3 +202,183 @@ func TestDeletePod(t *testing.T) {
 	}
 	npMgr.Unlock()
 }
+
+func TestAddHostNetworkPod(t *testing.T) {
+	npMgr := &NetworkPolicyManager{
+		nsMap:            make(map[string]*namespace),
+		podMap:           make(map[string]string),
+		TelemetryEnabled: false,
+	}
+
+	allNs, err := newNs(util.KubeAllNamespacesFlag)
+	if err != nil {
+		panic(err.Error)
+	}
+	npMgr.nsMap[util.KubeAllNamespacesFlag] = allNs
+
+	ipsMgr := ipsm.NewIpsetManager()
+	if err := ipsMgr.Save(util.IpsetTestConfigFile); err != nil {
+		t.Errorf("TestAddHostNetworkPod failed @ ipsMgr.Save")
+	}
+
+	defer func() {
+		if err := ipsMgr.Restore(util.IpsetTestConfigFile); err != nil {
+			t.Errorf("TestAddHostNetworkPod failed @ ipsMgr.Restore")
+		}
+	}()
+
+	podObj := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: "test-namespace",
+			Labels: map[string]string{
+				"app": "test-pod",
+			},
+		},
+		Status: corev1.PodStatus{
+			Phase: "Running",
+			PodIP: "1.2.3.4",
+		},
+		Spec: corev1.PodSpec{
+			HostNetwork: true,
+		},
+	}
+
+	npMgr.Lock()
+	if err := npMgr.AddPod(podObj); err != nil {
+		t.Errorf("TestAddHostNetworkPod failed @ AddPod")
+	}
+
+	if len(npMgr.podMap) >= 1 {
+		t.Errorf("TestAddHostNetworkPod failed @ podMap length check")
+	}
+	npMgr.Unlock()
+}
+
+func TestUpdateHostNetworkPod(t *testing.T) {
+	npMgr := &NetworkPolicyManager{
+		nsMap:            make(map[string]*namespace),
+		podMap:           make(map[string]string),
+		TelemetryEnabled: false,
+	}
+
+	allNs, err := newNs(util.KubeAllNamespacesFlag)
+	if err != nil {
+		panic(err.Error)
+	}
+	npMgr.nsMap[util.KubeAllNamespacesFlag] = allNs
+
+	ipsMgr := ipsm.NewIpsetManager()
+	if err := ipsMgr.Save(util.IpsetTestConfigFile); err != nil {
+		t.Errorf("TestUpdateHostNetworkPod failed @ ipsMgr.Save")
+	}
+
+	defer func() {
+		if err := ipsMgr.Restore(util.IpsetTestConfigFile); err != nil {
+			t.Errorf("TestUpdateHostNetworkPod failed @ ipsMgr.Restore")
+		}
+	}()
+
+	// HostNetwork check is done on the oldPodObj,
+	// so intentionally not adding hostnet true in newPodObj
+	oldPodObj := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "old-test-pod",
+			Namespace: "test-namespace",
+			Labels: map[string]string{
+				"app": "old-test-pod",
+			},
+		},
+		Status: corev1.PodStatus{
+			Phase: "Running",
+			PodIP: "1.2.3.4",
+		},
+		Spec: corev1.PodSpec{
+			HostNetwork: true,
+		},
+	}
+
+	newPodObj := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "new-test-pod",
+			Namespace: "test-namespace",
+			Labels: map[string]string{
+				"app": "new-test-pod",
+			},
+		},
+		Status: corev1.PodStatus{
+			Phase: "Running",
+			PodIP: "4.3.2.1",
+		},
+	}
+
+	npMgr.Lock()
+	if err := npMgr.AddPod(oldPodObj); err != nil {
+		t.Errorf("TestUpdateHostNetworkPod failed @ AddPod")
+	}
+
```

Comment:
Do we want to check the ipset situation? When adding oldPodObj, ipset should have no item for this pod but after adding new pod, it will have 1 ipset. An example, https://github.com/Azure/azure-container-networking/blob/master/npm/nwpolicy_test.go#L116
However, I didn't evaluate whether we can start a ipset manager here. If we can't start ipset mgr, then we don't have a way to check it. The current test is good enough.

Reply:
We will return early in the Update func after checking the hostnetwork true flag.

Reply:
OK I see.